### PR TITLE
add typings

### PR DESCRIPTION
**Pull request by yoyomo**
a Typescript file for typescript users

### CODE REVIEW

File: typings/message-bus.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+declare interface MessageBus {
```
Comment (benlangfeld) [inline typings/message-bus.d.ts]: How do we keep this in sync with both the implementation and the documentation? Is there tooling for checking that, or some way we can work it into our tests? Alternatively, is it possible to include this inline with the existing implementation and parse it out, such that it’s easier to spot when method signatures change but the type definition does not?

Comment (benlangfeld) [inline typings/message-bus.d.ts]: If this were merged, how would someone use it?

**Comment by SamSaffron** [comment]
I agree with Ben's concerns here, I think the most compelling argument here is just to have the original message bus in typescript and during build compile it to js. Then allow people to either import .ts or .js. I am not categorically against this but it is an involved change and we need to prove that the js generated is exactly the same as the current js file.

**Comment by ZogStriP** [comment]
Closing this. @yoyomo Feel free to re-open another PR if you port message bus to TS.

**Pull request closed without merging**